src/util/threadpool.rs

```diff
@@ -280,7 +272,21 @@ impl<Q: ScheduleQueue<T>, T: Debug> TaskPool<Q, T> {
     }
 
     fn pop_task(&mut self) -> Option<Task<T>> {
-        self.task_queue.pop()
+        let task = self.task_queue.pop();
+        if task.is_none() {
+            // try fill queue when queue is empty.
+            self.try_fill_queue();
```

Comment:
why not try receiving the task and return here directly, why do push and then pop?